### PR TITLE
feat(locksmith): Add support for buying multiple keys via stripe

### PR DESCRIPTION
**Pull request by searchableguy**
# Description

This PR adds support for buying multiple keys via stripe checkout. 

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/src/operations/lockOperations.ts [hunk 1]
```diff
@@ -115,7 +115,8 @@ export async function updateLockMigrationsLog(
 
 export async function isSoldOut(
   address: string,
-  chain: number
+  chain: number,
+  keysNeeded = 10
```
Comment (kalidiagne) [inline locksmith/src/operations/lockOperations.ts:119]: @searchableguy default value for `keysNeeded` shout not be equal to zero?

**Review by kalidiagne — APPROVED** [review]

File: locksmith/migrations/20220420120142-add-recipients-field-to-charge.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+module.exports = {
```
Comment (julien51) [inline locksmith/migrations/20220420120142-add-recipients-field-to-charge.js:3]: I think we should not use "charges" anymore, but that's fine for this PR.

File: locksmith/migrations/20220420111523-add-recipients-field-to-payment-intent.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+'use strict'
+
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.addColumn('PaymentIntents', 'recipients', {
+      type: Sequelize.ARRAY(Sequelize.STRING),
```
Comment (julien51) [inline locksmith/migrations/20220420111523-add-recipients-field-to-payment-intent.js:6]: Let's use the `Sequelize.JSON` instead for compliance.

**Review by julien51 — APPROVED** [review]
Outside of the quick data model change I think it's looking good!

File: locksmith/migrations/20220420120142-add-recipients-field-to-charge.js (hunk 2 above)
Comment (searchableguy) [reply locksmith/migrations/20220420120142-add-recipients-field-to-charge.js:3]: We do use it to save the details so I adapted it.

File: locksmith/src/operations/lockOperations.ts (hunk 1 above)
Comment (searchableguy) [reply locksmith/src/operations/lockOperations.ts:119]: The default is high to provide a reasonable estimation of *are there enough keys for me to buy?*

File: locksmith/migrations/20220420111523-add-recipients-field-to-payment-intent.js (hunk 3 above)
Comment (searchableguy) [reply locksmith/migrations/20220420111523-add-recipients-field-to-payment-intent.js:6]: okay

**Pull request merged**